crates/runner/src/vm/mod.rs

```diff
@@ -23,13 +51,146 @@ impl State {
         }
     }
 }
+/// Raw program data structure as it appears in the compilation output JSON.
+///
+/// Array of instructions, where each instruction is 4 hexadecimal strings:
+/// [opcode, arg1, arg2, arg3]
+#[derive(Debug, Deserialize)]
+struct ProgramRaw {
+    data: Vec<[String; 4]>,
```

Comment:
shouldn't we use `#[repr(C)]` to load directly a .bin file as a list of `[u32; 4]`?